### PR TITLE
Add chart-operator app CR

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward https://github.com/giantswarm/giantswarm/issues/6078
Adding `chart-operator` app CR and deleting resources that we are delegating to app-operator. 

`clusterapi` PR would follow.

### CODE REVIEW

File: pkg/v20/key/key.go [hunk 1]
```diff
@@ -63,10 +63,15 @@ func ClusterOrganization(clusterGuestConfig v1alpha1.ClusterGuestConfig) string
 }
 
 // CommonAppSpecs returns apps installed for all providers.
-// Note: When adding app specs you also need to add the chart name to the
```
Comment (tomahawk28) [inline pkg/v20/key/key.go]: Since this is copy & paste from other resources, I'm going to delete this part.

**Comment by tomahawk28** [comment]
Tested on `ginger`, created new cluster. No issue found. all `apps` are running with status update.

File: service/controller/aws/v20/resource_set.go [hunk 2]
```diff
@@ -23,13 +24,11 @@ import (
...
 	chartconfigservice "github.com/giantswarm/cluster-operator/pkg/v20/chartconfig"
 	configmapservice "github.com/giantswarm/cluster-operator/pkg/v20/configmap"
+	"github.com/giantswarm/cluster-operator/pkg/v20/resource/app"
 	"github.com/giantswarm/cluster-operator/pkg/v20/resource/certconfig"
-	"github.com/giantswarm/cluster-operator/pkg/v20/resource/chartoperator"
 	"github.com/giantswarm/cluster-operator/pkg/v20/resource/clusterconfigmap"
 	"github.com/giantswarm/cluster-operator/pkg/v20/resource/encryptionkey"
 	"github.com/giantswarm/cluster-operator/pkg/v20/resource/kubeconfig"
-	"github.com/giantswarm/cluster-operator/pkg/v20/resource/namespace"
-	"github.com/giantswarm/cluster-operator/pkg/v20/resource/tiller"
```
Comment (xh3b4sd) [inline service/controller/aws/v20/resource_set.go]: Why do we also delete these resources when the PR is only about `chart-operator`?

**Comment by rossf7** [comment]
> Tested on ginger, created new cluster. No issue found. all apps are running with status update.

👍 Ack. But it also needs to be tested once the resources are removed.

**Comment by tomahawk28** [comment]
@rossf7 I'm only making changes on general `pkg`, that 3 resources you mentioned are deleted already in there. 
`clusterapi` changes would follow on next PR.

File: service/controller/aws/v20/resource_set.go (hunk 2 above)
Comment (tomahawk28) [reply service/controller/aws/v20/resource_set.go]: As Ross mentioned in this comment, https://github.com/giantswarm/cluster-operator/pull/644#pullrequestreview-282612439 we need a proper test on when `chart-operator` resource got removed. We already move these parts to app-operator, would like to see that it's going well w/o chart-operator and other resources. (namespace, tiller)

**Pull request merged**